webapp/packages/plugin-holidays/src/HolidaysService.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { importLazyComponent } from '@cloudbeaver/core-blocks';
+import { Bootstrap, injectable } from '@cloudbeaver/core-di';
+import { AdministrationTopAppBarService } from '@cloudbeaver/plugin-administration';
+import { TopNavService } from '@cloudbeaver/plugin-top-app-bar';
+
+import type { IHoliday } from './IHoliday.js';
+
+const HolidayButton = importLazyComponent(() => import('./HolidayActionButton.js').then(m => m.HolidayActionButton));
+
+@injectable()
+export class HolidaysService extends Bootstrap {
+  private readonly holidays: IHoliday[] = [];
+
+  constructor(
+    private readonly topNavService: TopNavService,
+    private readonly administrationTopAppBarService: AdministrationTopAppBarService,
+  ) {
+    super();
+  }
+
+  override register() {
+    this.topNavService.placeholder.add(HolidayButton, 4);
+    this.administrationTopAppBarService.placeholder.add(HolidayButton, 4);
+  }
```

Comment:
same situation here, this service depends both on public and administration part, it needs to be split